com_migratetojoomla/admin/src/controller/MainController.php

```diff
@@ -0,0 +1,278 @@
+     */
+    public  function downloadMedia($data = [])
+    {
+        $app   = Factory::getApplication();
+        $method = $data['mediaoptions'];
+        $source = '';
+
+        switch ($method) {
+            case 2:
+                $source = $data['basedir'];
+                break;
+            case 3:
+                $this->mediaDownloadManager = new FtpHelper($data);
+                $response = $this->mediaDownloadManager->login();
+                $source = $data['ftpbasedir'];
+                break;
+            case 1:
+            default:
+                $this->mediaDownloadManager = new HttpHelper($data['livewebsiteurl']);
+                $source = $data['livewebsiteurl'];
+                break;
+        }
+
+        $source = MainHelper::addTrailingSlashit($source) . 'wp-content\uploads';
+        $destination = MainHelper::addTrailingSlashit(JPATH_ROOT) . 'images';
+
+        try {
+            if ($method == 2) {
+                Folder::copy($source, $destination, '', true, false);
+            } else {
+                $this->copy($source, $destination);
+            }
+            $app->enqueueMessage(TEXT::_('COM_MIGRATETOJOOMLA_DOWNLOAD_MEDIA_SUCCESSFULLY'), 'success');
+        } catch (\RuntimeException $th) {
+            $app->enqueueMessage(TEXT::_('COM_MIGRATETOJOOMLA_DOWNLOAD_MEDIA_UNSUCCESSFULLY'), 'danger');
+        }
+    }
+
+    /**
+     * Method to copy a file or a directory
+     *
+     * @param string $source Original file or directory name
+     * @param string $destination Destination file or directory name
+     * @param bool $recursive Recursive copy?
+     * @return bool File copied or not
+     * 
+     * @since  1.0
+     */
+    public function copy($source, $destination)
+    {
+        if ($this->mediaDownloadManager->isDir($source)) {
+            // Directory
+            return $this->copyDir($source, $destination);
+        } else {
+            // File
+            return $this->copyFile($source, $destination);
+        }
+    }
+
+    /**
+     * Method to copy file
+     * 
+     * @param string $source source path
+     * @param string $destination destination path
+     * 
+     * @return boolean True on success
+     * 
+     * @since  1.0
+     */
+    public function copyFile($source, $destination)
+    {
+        $response = false;
+        if (file_exists($destination) && (filesize($destination) > 0)) {
+            // file Already downloaded 
+            return true;
+        }
+
+        $response = $this->mediaDownloadManager->getContent($source, $destination);
+        return $response;
+    }
+
+    /**
+     * Method to make directory and copy it's content
+     * 
+     * @param string $source Source path
+     * @param string $source Destination path
+     * 
+     * @return boolean True on Success
+     * 
+     * @since  1.0
+     */
+    public function copyDir($source, $destination)
+    {
+        $response = true;
+        if (!is_dir($destination)) {
```

Comment:
When it's a file, you have a problem.

Reply:
copy Dir only call for a directory not file